addons/website/static/src/components/survey/survey.js

```diff
@@ -0,0 +1,514 @@
+    }).catch((_) => false);
+
+    const allPalettes = {}, palettes = {};
+    const style = window.getComputedStyle(document.documentElement);
+    const themeNames = weUtils.getCSSVariableValue('theme-names', style).split(' ');
+    themeNames.forEach((rawName) => {
+        const name = rawName.replace(/'/g, '');
+        const count = weUtils.getCSSVariableValue(`o-count-${name}`, style);
+        for (let i = 1; i <= count; i++) {
+            const paletteName = `${name}-${i}`;
+            const palette = {};
+            palette['id'] = paletteName;
+            for (let j = 1; j <= 5; j += 1) {
+                const color = weUtils.getCSSVariableValue(`o-palette-${name}-${i}-o-color-${j}`, style);
+                palette[`color${j}`] = color;
+            }
+            let duplicate = false;
+            for (const validatedPalette of Object.values(allPalettes)) {
+                if (validatedPalette.color1.toLowerCase() === palette.color1.toLowerCase() && validatedPalette.color2.toLowerCase() === palette.color2.toLowerCase()) {
+                    duplicate = true;
+                }
+            }
+            if (!duplicate) {
+                allPalettes[paletteName] = palette;
+                if (Object.keys(palettes).length < MAX_PALETTES) {
+                    palettes[paletteName] = allPalettes[paletteName];
+                }
+            }
+        }
+    });
+
+    const vals = await Promise.all([features, logo, industries]);
+    if (!vals[2]) {
+        await skipSurvey(parseInt(wid));
+        return false;
+    }
+
+    return {
+        selectedType: undefined,
+        selectedPurpose: undefined,
+        selectedIndustry: undefined,
+        industries: vals[2],
+        logo: vals[1],
+        selectedPalette: undefined,
+        recommendedPalette: undefined,
+        palettes: palettes,
+        allPalettes: allPalettes,
+        features: vals[0],
+        themes: Array(3).fill({
+            name: undefined,
+            id: undefined,
+            svg: undefined,
+        }),
+    };
+}
+
+async function makeStore(wid) {
+    const state = await getInitialState(wid);
+    if (!state) {
+        return false;
+    }
+    const store = new Store({state, actions, getters});
+    return store;
+}
+
+async function makeEnvironment() {
+    const env = {};
+    env.router = new Router(env, ROUTES);
+    await env.router.start();
+    const store = await makeStore(env.router.currentParams.wid);
+    if (!store) {
+        return false;
+    }
+    env.store = store;
+    const qweb = new QWeb({translateFn: _t});
+    const loaderTemplate = await owl.utils.loadFile('/website/static/src/xml/theme_preview.xml');
+    const surveyTemplates = await owl.utils.loadFile('/website/static/src/components/survey/survey.xml');
+    qweb.addTemplates(loaderTemplate);
+    qweb.addTemplates(surveyTemplates);
+    env.qweb = qweb;
+    return env;
+}
+
+async function setup() {
+    const env = await makeEnvironment();
```

Comment:
when can this fail, and if it does, how is it handled?

Reply:
call to IAP for industry fetching could fail. In this case we redirect the user to the 25 themes selection screen. With the` if (env) ` I avoid to mount the App and to show the first survey screen during the delay between the launch of the redirection and the actual redirection. But this probably not the best way to do it.

Reply:
I think I made this comment between pushes: it's now very explicit that you're skipping the survey when no industries are found :+1:

Reply:
Yes I made some changes yesterday. It looks better like that